### PR TITLE
fix: set Path as empty if path is not specified for a source in multiple sources (#11756)

### LINKED ISSUE #11756: 2.6 RC1: App summary path in multi-source app shows revision instead
Checklist:

**Describe the bug**

If I look at an Application's summary, the `path` field is populated with the revision link instead of the path.

**To Reproduce**

Use this app: 

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: test
  namespace: argocd
spec:
  destination:
    namespace: default
    server: https://kubernetes.default.svc
  project: default
  sources:
  - ref: elastic
    repoURL: https://github.com/crenshaw-dev/values-file-repo
  - helm:
      valueFiles:
      - $elastic/values-prod.yaml
    path: helm-guestbook
    repoURL: https://github.com/argoproj/argocd-example-apps
    targetRevision: HEAD
```

View the app in the UI and go to the app summary page.

**Expected behavior**

I think leaving the path field empty would be appropriate for this app. The source has no path, because it's acting only as a source for the values file, and not as sources to be rendered.

**Screenshots**

image

**Version**

2.6.0-rc1

### PR DESCRIPTION
**Pull request by ishitasequeira**
Fixes: #11756 

After changes:
image

Signed-off-by: ishitasequeira <ishiseq29@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: ui/src/app/shared/components/revision.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
-    revision = revision || '';
+export const Revision = ({repoUrl, revision, path, title, children}: {repoUrl: string; revision: string; path?: string; title?: string; children?: React.ReactNode}) => {
     const hasPath = path && path !== '.';
+    if (title == 'PATH' && !hasPath) {
+        return <span></span>
+    }
```
[the remark this answers is not in the document]
Comment (ishitasequeira) [reply ui/src/app/shared/components/revision.tsx:8]: The value can also be `undefined`, will set it to just `title == 'PATH' && path`

File: ui/src/app/shared/components/revision.tsx [hunk 2]
```diff
@@ -1,9 +1,12 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
-    revision = revision || '';
+export const Revision = ({repoUrl, revision, path, title, children}: {repoUrl: string; revision: string; path?: string; title?: string; children?: React.ReactNode}) => {
     const hasPath = path && path !== '.';
+    if (title == 'PATH' && path) {
+        return <span></span>
+    }
```
Comment (crenshaw-dev) [inline ui/src/app/shared/components/revision.tsx:8]:
```suggestion
    if (title == 'PATH' && !path) {
        // This source literally has no path, so we won't show one.
        return <span></span>
    }
    const hasPath = path && path !== '.';
```

Just to clean up.

Do we have to pass `title`?

Comment (ishitasequeira) [reply ui/src/app/shared/components/revision.tsx:8]: title is just to distinguish the field that is calling the Revision tag. The same Revision tag is getting used to return the value of TargetRevision which includes path in generating the URL. So, if we don't specify `title`, the value of `TargetRevision` also gets set empty.

Comment (crenshaw-dev) [reply ui/src/app/shared/components/revision.tsx]: Ah, sure. Can we pass a boolean instead? `isForRevision: false`, or something like that?

**Comment by crenshaw-dev** [comment]
@ishitasequeira can you fix the UI lint error?

File: ui/src/app/shared/components/revision.tsx [hunk 3]
```diff
@@ -1,7 +1,11 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
+export const Revision = ({repoUrl, revision, path, isForPath, children}: {repoUrl: string; revision: string; path?: string; isForPath?: boolean; children?: React.ReactNode}) => {
+    if (isForPath && path) {
```
Comment (crenshaw-dev) [inline ui/src/app/shared/components/revision.tsx:5]:
```suggestion
    if (isForPath && !path) {
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Review by crenshaw-dev — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.6

**Pull request merged**